rama-net/src/tls/server/config.rs

```diff
@@ -118,6 +128,69 @@ pub struct ServerAuthData {
     pub ocsp: Option<Vec<u8>>,
 }
 
+#[derive(Clone)]
+/// Dynamic issuer which internally contains the dyn issuer
+pub struct DynamicIssuer {
+    /// Issuer not public in case we want to migrate away from dyn approach to alternative (eg channels)
+    issuer: Arc<dyn DynDynamicCertIssuer + Send + Sync>,
+}
+
+impl DynamicIssuer {
+    pub fn new<T: DynamicCertIssuer + Send + Sync + 'static>(issuer: T) -> Self {
+        return Self {
+            issuer: Arc::new(issuer),
+        };
+    }
+
+    pub async fn issue_cert(
+        &self,
+        client_hello: ClientHello,
+        server_name: Option<Host>,
+    ) -> Result<ServerAuthData, OpaqueError> {
+        self.issuer.issue_cert(client_hello, server_name).await
+    }
+}
+
+impl std::fmt::Debug for DynamicIssuer {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("DynamicIssuer").finish()
+    }
+}
+
+/// Trait that needs to be implemented by cert issuers to support dynamically
+/// issueing (external) certs based on client_hello input.
+pub trait DynamicCertIssuer {
+    fn issue_cert(
+        &self,
+        client_hello: ClientHello,
+        server_name: Option<Host>,
+    ) -> impl Future<Output = Result<ServerAuthData, OpaqueError>> + Send + Sync + '_;
+}
+
+/// Internal trait to support dynamic dispatch of trait with async fn.
+/// See trait DynService<S, Request> for more info about this pattern.
```

Comment:
Can you link to it using markdown links, you can do so using rust mod syntax etc